diff --git a/node_modules/jsesc/jsesc.js b/node_modules/jsesc/jsesc.js
index 7b8b923..d63726d 100644
--- a/node_modules/jsesc/jsesc.js
+++ b/node_modules/jsesc/jsesc.js
@@ -30,7 +30,7 @@ const forEach = (array, callback) => {
 
 const toString = object.toString;
 const isArray = Array.isArray;
-const isBuffer = Buffer.isBuffer;
+const isBuffer = global.Buffer ? Buffer.isBuffer : require('buffer').Buffer.isBuffer;
 const isObject = (value) => {
 	// This is a very simple check, but it’s good enough for what we need.
 	return toString.call(value) == '[object Object]';
